sdk/identity/identity/samples/ClientSideUserAuthentication.md

```diff
@@ -0,0 +1,62 @@
+# Client side user authentication
+
+Client side applications often need to authenticate users to interact with resources in Azure. Some examples of this might be a command line tool which fetches secrets a user has access to from a key vault to setup a local test environment, or a GUI based application which allows a user to browse storage blobs they have access to. This sample demonstrates authenticating users with the `@azure/identity` library.
+
+## Interactive user authentication
+
+Most often authenticating users requires some user interaction. Properly handling this user interaction for OAuth2 authorization code or device code authentication can be challenging. To simplify this for client side applications the `@azure/identity` library provides the `InteractiveBrowserCredential` and the `DeviceCodeCredential`. These credentials are designed to handle the user interactions needed to authenticate via these two client side authentication flows, so the application developer can simply create the credential and authenticate clients with it.
+
+## Authenticating users with the InteractiveBrowserCredential
+
+For clients which have a default browser available, the `InteractiveBrowserCredential` provides the most simple user authentication experience. In the sample below an application authenticates a `KeyClient` (from [@azure/keyvault-keys](https://www.npmjs.com/package/@azure/keyvault-keys)) using the `InteractiveBrowserCredential`.
+
+```ts
+const { InteractiveBrowserCredential } = require("@azure/identity");
+const { KeyClient } = require("@azure/keyvault-keys");
+
+async function main() {
+  const credential = new InteractiveBrowserCredential({
+    // By default, tenantId will be "organizations". You might assign a specific tenant this way.
+    tenantId: process.env.AZURE_TENANT_ID,
+    // By default, clientId will be the same used by the Azure CLI. You might assign a specific client ID this way.
+    clientId: process.env.AZURE_CLIENT_ID  
+  });
+
+  const keyVaultUrl = `https://key-vault-name.vault.azure.net`;
+  const client = new KeyClient(keyVaultUrl, credential);
+
+  // Retrieving the properties of the existing keys in that specific Key Vault.
+  console.log(await client.listPropertiesOfKeys().next());
+}
+
+main().then(console.log).catch((e) => console.error(e));
+```
+
+As code uses the `SecretClient` in the above sample, the `InteractiveBrowserCredential` will automatically authenticate the user by launching the default system browser prompting the user to login. In this case the user interaction happens on demand as is necessary to authenticate calls from the client.
```

Comment:
"In this case the user interaction happens on demand as is necessary to authenticate calls from the client." feels awkward.

One possible change: "Once the user successfully logs in, the user can then call into Azure services."